src/Compilers/Core/CodeAnalysisTest/Diagnostics/SystemLanguageAnalyzer.cs

```diff
@@ -128,4 +132,118 @@ void Report(CompilationAnalysisContext context, IFieldSymbol field, DiagnosticDe
             context.ReportDiagnostic(Diagnostic.Create(descriptor, field.Locations.FirstOrDefault()));
         }
     }
+
+    /// <summary>Analyzer used to identify local variables that could be declared Const.</summary>
+    public class LocalCouldBeConstAnalyzer : DiagnosticAnalyzer
```

Comment:
Is this really a test-only analyzer or is it better to move it to roslyn-analyzers repo in future? Seems like a useful analyzer to me.
I wonder if you are adding it here only because there is no other way to currently write an analyzer that targets new Roslyn analyzer APIs.